examples/ristretto/src/ristretto.rs

```diff
@@ -0,0 +1,177 @@
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8,
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xedu8
+    ))
+}
+
+fn D() -> FieldElement {
+    FieldElement::from_byte_seq_be(&byte_seq!(
+        0x52u8, 0x03u8, 0x6cu8, 0xeeu8, 0x2bu8, 0x6fu8, 0xfeu8, 0x73u8, 0x8cu8, 0xc7u8, 0x40u8,
+        0x79u8, 0x77u8, 0x79u8, 0xe8u8, 0x98u8, 0x00u8, 0x70u8, 0x0au8, 0x4du8, 0x41u8, 0x41u8,
+        0xd8u8, 0xabu8, 0x75u8, 0xebu8, 0x4du8, 0xcau8, 0x13u8, 0x59u8, 0x78u8, 0xa3u8
+    ))
+}
+
+fn SQRT_M1() -> FieldElement {
+    FieldElement::from_byte_seq_be(&byte_seq!(
+        0x2bu8, 0x83u8, 0x24u8, 0x80u8, 0x4fu8, 0xc1u8, 0xdfu8, 0x0bu8, 0x2bu8, 0x4du8, 0x00u8,
+        0x99u8, 0x3du8, 0xfbu8, 0xd7u8, 0xa7u8, 0x2fu8, 0x43u8, 0x18u8, 0x06u8, 0xadu8, 0x2fu8,
+        0xe4u8, 0x78u8, 0xc4u8, 0xeeu8, 0x1bu8, 0x27u8, 0x4au8, 0x0eu8, 0xa0u8, 0xb0u8
+    ))
+}
+
+fn INVSQRT_A_MINUS_D() -> FieldElement {
+    FieldElement::from_byte_seq_be(&byte_seq!(
+        0x78u8, 0x6cu8, 0x89u8, 0x05u8, 0xcfu8, 0xafu8, 0xfcu8, 0xa2u8, 0x16u8, 0xc2u8, 0x7bu8,
+        0x91u8, 0xfeu8, 0x01u8, 0xd8u8, 0x40u8, 0x9du8, 0x2fu8, 0x16u8, 0x17u8, 0x5au8, 0x41u8,
+        0x72u8, 0xbeu8, 0x99u8, 0xc8u8, 0xfdu8, 0xaau8, 0x80u8, 0x5du8, 0x40u8, 0xeau8
+    ))
+}
+
+fn SQRT_AD_MINUS_ONE() -> FieldElement {
+    FieldElement::from_byte_seq_be(&byte_seq!(
+        0x37u8, 0x69u8, 0x31u8, 0xbfu8, 0x2bu8, 0x83u8, 0x48u8, 0xacu8, 0x0fu8, 0x3cu8, 0xfcu8,
+        0xc9u8, 0x31u8, 0xf5u8, 0xd1u8, 0xfdu8, 0xafu8, 0x9du8, 0x8eu8, 0x0cu8, 0x1bu8, 0x78u8,
+        0x54u8, 0xbdu8, 0x7eu8, 0x97u8, 0xf6u8, 0xa0u8, 0x49u8, 0x7bu8, 0x2eu8, 0x1bu8
+    ))
+}
+
+fn ONE_MINUS_D_SQ() -> FieldElement {
+    FieldElement::from_byte_seq_be(&byte_seq!(
+        0x02u8, 0x90u8, 0x72u8, 0xa8u8, 0xb2u8, 0xb3u8, 0xe0u8, 0xd7u8, 0x99u8, 0x94u8, 0xabu8,
+        0xddu8, 0xbeu8, 0x70u8, 0xdfu8, 0xe4u8, 0x2cu8, 0x81u8, 0xa1u8, 0x38u8, 0xcdu8, 0x5eu8,
+        0x35u8, 0x0fu8, 0xe2u8, 0x7cu8, 0x09u8, 0xc1u8, 0x94u8, 0x5fu8, 0xc1u8, 0x76u8
+    ))
+}
+
+fn D_MINUS_ONE_SQ() -> FieldElement {
+    FieldElement::from_byte_seq_be(&byte_seq!(
+        0x59u8, 0x68u8, 0xb3u8, 0x7au8, 0xf6u8, 0x6cu8, 0x22u8, 0x41u8, 0x4cu8, 0xdcu8, 0xd3u8,
+        0x2fu8, 0x52u8, 0x9bu8, 0x4eu8, 0xebu8, 0xd2u8, 0x9eu8, 0x4au8, 0x2cu8, 0xb0u8, 0x1eu8,
+        0x19u8, 0x99u8, 0x31u8, 0xadu8, 0x5au8, 0xaau8, 0x44u8, 0xedu8, 0x4du8, 0x20u8
+    ))
+}
+
+// === Helper functions === //
+
+// Creates a field element from the given literal.
+fn fe(x: usize) -> FieldElement {
+    FieldElement::from_literal(x as u128)
+}
+
+// === Internal Utility Functions === //
+
+// Checks if a given field element is negative. A negative field element is defined as an odd number.
+fn is_negative(e: FieldElement) -> bool {
+    e % fe(2) == fe(1)
+}
+
+// Checks if two given field elements are equal.
+fn ct_eq(u: FieldElement, v: FieldElement) -> bool {
```

Comment:
Can we not call these three function `ct_` when they are not?

Reply:
Also don't worry about the delay, thanks for reviewing

Reply:
It's public integers, so they're not constant time.
The secret integer versions aren't completely finished yet I think. They'd have constant time comparison functions.
You can add a comment saying that these functions must be implemented in a constant time manner.

Reply:
Yes, to adding the comment, but no to keeping the `ct_` 🙂

Reply:
Should be done now